WcaOnRails/app/controllers/git_hub_controller.rb

```diff
@@ -0,0 +1,22 @@
+class GitHubController < ApplicationController
+  def edit_translation
+  end
+
+  def update_translation
+    user_login = Octokit.user.login
+    origin_repo = "#{user_login}/worldcubeassociation.org"
+    upstream_repo = "thewca/worldcubeassociation.org"
+    content = params[:translation][:content].gsub(/\r/, '') # We don't want \r characters, but browsers add them automatically.
+    locale = params[:translation][:locale]
+    file_path = "WcaOnRails/config/locales/#{locale}.yml"
+    message = "Update #{locale} translation."
+    content_digest = Digest::SHA1.hexdigest(content)
+    branch_name = "translation-#{content_digest}"
+    upstream_sha = Octokit.ref(upstream_repo, "heads/master")[:object][:sha]
+    Octokit.create_ref(origin_repo, "heads/#{branch_name}", upstream_sha)
+    current_content_sha = Octokit.content(origin_repo, path: file_path, ref: branch_name)[:sha]
+    Octokit.update_content(origin_repo, file_path, message, current_content_sha, content, branch: branch_name)
+    pr_url = Octokit.create_pull_request(upstream_repo, "master", "#{user_login}:#{branch_name}", message)[:html_url]
+    redirect_to pr_url
```

Comment:
I found it handy to redirect to the PR url, as it was so simple. I imagine this could surprise translators without any GitHub experience, so it can also be a friendly page with some "Thank you!" and link to the PR (+ random sweet cat image as an award? ^^).